src/react/components/connect.tsx

```diff
@@ -3,35 +3,38 @@ import * as React from "react"
 import shallowEqual from "../../utils/shallowEqual"
 import propValidation from "../../utils/propsValidation"
 
-export default function connect(mapToProps) {
-  return Child =>
-    class Connected extends React.Component {
-      static contextTypes = {
-        store: propValidation
-      }
-      unsubscribe
-      state = this.getProps()
-      componentWillMount() {
-        this.unsubscribe = this.context.store.subscribe(this.update)
-      }
-      componentWillUnmount() {
-        this.unsubscribe(this.update)
-      }
-      getProps() {
-        const state =
-          (this.context.store && this.context.store.getState()) || {}
-        return mapToProps(state, this.props)
-      }
-      update = () => {
-        const mapped = this.getProps()
-        if (!shallowEqual(mapped, this.state)) {
-          this.setState(mapped)
-        }
-      }
-      render() {
-        return (
-          <Child store={this.context.store} {...this.props} {...this.state} />
-        )
-      }
+export class Connect extends React.Component<any> {
+  static contextTypes = {
+    store: propValidation
+  }
+  unsubscribe
+  state = this.getProps()
+  componentWillMount() {
+    this.unsubscribe = this.context.store.subscribe(this.update)
+  }
+  componentWillUnmount() {
+    this.unsubscribe(this.update)
+  }
+  getProps() {
+    const { mapToProps } = this.props
+    const state = (this.context.store && this.context.store.getState()) || {}
+    return mapToProps(state, this.props)
+  }
+  update = () => {
+    const mapped = this.getProps()
+    if (!shallowEqual(mapped, this.state)) {
+      this.setState(mapped)
     }
+  }
+  render() {
+    return this.props.children(this.state)
+  }
+}
+
+export default function connect(mapToProps) {
```

Comment:
Also, what you think about having two files, one connect.ts and another one, Connect.ts?
In this project, we're trying to have one export per file, as a pattern.

Reply:
I actually wrote that way at first and then moved it back into connect.tsx.  I can move it back into it's own file if you'd like.

Reply:
Yes, please do that :)

Reply:
What do you want to call the two files?  The linter complaines with two files named `Connect.tsx` and `connect.tsx`.  Would you be okay with something like `Connect.tsx` as the Component w/ render callback and `connect-hoc.tsx` or `ConnectHOC.tsx`.  This wouldn't impact the imports from the library.

Reply:
Didn't know about that as well, so better to leave the way it is.